tracer/src/Datadog.Trace/AppSec/AppSecRequestContext.cs

```diff
@@ -35,6 +36,16 @@ internal void CloseWebSpan(TraceTagCollection tags, Span span)
             {
                 span.SetMetaStruct(StackKey, MetaStructHelper.ObjectToByteArray(_raspStackTraces));
             }
+
+            _raspTelemetryHelper?.GenerateRaspSpanMetricTags(span.Tags);
```

Comment:
I assume not enter the lock here is by design? Because it's a read operation, we don't care that another thread may be modify the values as they're being read?

Reply:
This statement is inside a lock block that starts at the beginning of the method. The operation should be thread safe.